typing/cmi_format.ml

```diff
@@ -20,8 +20,7 @@ type pers_flags =
   | Unsafe_string
 
 type error =
-    Not_an_interface of string
-  | Wrong_version_interface of string * string
+    Not_an_interface of string * string option
```

Comment:
You could just keep the structured error (`Magic_number.parse_error`) here.

Reply:
What about now?

Reply:
In the rebased version, I keep structured errors in exception payloads. (And the structured errors are nicer than before.)